### ISSUE TITLE
Version `2.1.0` silently broke `wasm` builds with `uuid`

### ISSUE THREAD

**Issue by ilslv** [mongodb/bson-rust#334]
Version `2.1.0` enables `uuid`s `v4` feature, which seems to be a breaking change, as [`juniper`](https://github.com/graphql-rust/juniper) `wasm` build [began failing today](https://github.com/graphql-rust/juniper/runs/4530682725?check_suite_focus=true#step:4:168).

**Comment by patrickfreed** [comment]
Hi @ilslv, thanks for reporting this issue! I can confirm that, by default, `bson` 2.1.0 doesn't build on wasm. It does appear that downstream crates can work around this for now by adding an explicit dependency on `uuid` with the `wasm-bindgen` feature flag enabled. I've also filed RUST-1130 to discuss with the team options for improving this on our end. Please let me know if the workaround works for you or if you have any more questions, thanks!

**Comment by ilslv** [comment]
Thanks for the workaround! 
Maybe we can add `wasm-bindgen` feature for `uuid` based on target right in this repo and also add `wasm` build on the CI? I'll be happy to contribute this.

**Comment by patrickfreed** [comment]
Yeah, we plan on doing something like that, and thanks for offering to contribute those changes @ilslv! Since our CI system is built in-house and is somewhat tricky to configure, we usually handle doing stuff like that ourselves though. That said, we plan to get to RUST-1130 sometime over the next quarter, so stay tuned for updates! In the meantime, I hope the above workaround works sufficiently well.
